instrumentation/liberty/liberty-dispatcher/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/dispatcher/LibertyDispatcherInstrumentationModule.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.liberty.dispatcher;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(InstrumentationModule.class)
+public class LibertyDispatcherInstrumentationModule extends InstrumentationModule {
```

Comment:
@trask should we have `classLoaderMatcher` here somewhere? Or `classLoaderOptimization`?

Reply:
I believe it is not needed because this one matches based on exact class name. `classLoaderOptimization` usually uses `hasClassesNamed` which uses `ClassLoader.getResource` to verify that given class loader can see some type that the instrumentation depends on. For example when you want to instrument all classes that implement some interface you first check whether current class loader can see that interface, if it doesn't the assumption is that this class loader can't contain classes that would be needed to instrumented by this module. The idea behind this is that checking whether class implements some interface is currently expensive so we try to avoid it by doing a cheaper check before it, but `ClassLoader.getResource` isn't free either so if it isn't really needed i would skip it.